x/auth/middleware/priority.go

```diff
@@ -50,12 +50,15 @@ func (h txPriorityHandler) SimulateTx(ctx context.Context, sdkTx sdk.Tx, req tx.
 	return h.next.SimulateTx(ctx, sdkTx, req)
 }
 
-// GetTxPriority returns a naive tx priority based on the total sum of all fees
+// GetTxPriority returns a naive tx priority based on the amount of the smallest denomination of the fee
 // provided in a transaction.
 func GetTxPriority(fee sdk.Coins) int64 {
 	var priority int64
 	for _, c := range fee {
-		priority += c.Amount.Int64()
+		p := c.Amount.Int64()
+		if priority == 0 || p < priority {
```

Comment:
i don't think we need the 0 check

```suggestion
		if p < priority {
```

Reply:
But then the priority will always be zero